lib/Frontend/FrontendOptions.cpp

```diff
@@ -454,6 +454,39 @@ bool FrontendOptions::doesActionProduceTextualOutput(ActionType action) {
   }
 }
 
+bool FrontendOptions::doesActionNeedSILOptimizer(ActionType action) {
+  switch (action) {
+  case ActionType::NoneAction:
+  case ActionType::Parse:
+  case ActionType::ResolveImports:
+  case ActionType::Typecheck:
+  case ActionType::DumpParse:
+  case ActionType::DumpInterfaceHash:
+  case ActionType::EmitSyntax:
+  case ActionType::DumpAST:
+  case ActionType::PrintAST:
+  case ActionType::DumpScopeMaps:
+  case ActionType::DumpTypeRefinementContexts:
+  case ActionType::EmitImportedModules:
+  case ActionType::EmitPCH:
+  case ActionType::EmitSILGen:
+    return false;
+  case ActionType::EmitSIL:
+  case ActionType::EmitModuleOnly:
+  case ActionType::MergeModules:
+  case ActionType::EmitSIBGen:
```

Comment:
If we're going to change behavior anyway, we may as well pull EmitSIBGen out of this list.